src/libNode/MicroBlockPostProcessing.cpp

```diff
@@ -138,7 +138,11 @@ bool Node::ProcessMicroblockConsensus(const vector<unsigned char>& message,
 void Node::CommitMicroBlockConsensusBuffer() {
   lock_guard<mutex> g(m_mutexMicroBlockConsensusBuffer);
 
-  for (const auto& i : m_microBlockConsensusBuffer[m_mediator.m_consensusID]) {
+  /// Call-by-reference MUST NOT be used here, because later "i" (vector
+  /// element) will be passed to another thread, and this vector might grow
+  /// (memory re-allocated) in a separated thread (ProcessMicroblockConsensus),
+  /// then racing-condition could happened under this kind of scenario
+  for (const auto i : m_microBlockConsensusBuffer[m_mediator.m_consensusID]) {
     auto runconsensus = [this, i]() {
```

Comment:
I think instead of capturing `i`, it has to be passed as a parameter

Reply:
Can do like this:
```
for (const auto& i ...) {
   auto runconsensus = [this, i]() {
      vector<unsigned char> msg = i.second;
      ProcessMicroblockConsensusCore(msg, ...)
```

Reply:
There should be another reason for the crash, cause the default behavior of lambda capturing should be "copy by value". Need to dig out more for investigating this issue.